app/src/main/java/doubleslash05/mini/team11/MainActivity.java

```diff
@@ -0,0 +1,333 @@
+
+import android.Manifest;
+import android.app.ProgressDialog;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+
+import com.icaksama.rapidsphinx.RapidCompletionListener;
+import com.icaksama.rapidsphinx.RapidPreparationListener;
+import com.icaksama.rapidsphinx.RapidSphinx;
+import com.icaksama.rapidsphinx.RapidSphinxListener;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.cmu.pocketsphinx.Config;
+
+public class MainActivity extends AppCompatActivity implements RapidSphinxListener {
+
+    public static final int sub = 1001;
+
+    // 이제 응용해봐야하는데..
+
+    private RapidSphinx rapidSphinx;
+    private Button btnRecognizer;
+
+
+    //    private Button btnStartAudio; 오디오 출력 필요없음.
+    private Button btnSync;
+    private EditText txtWords;
+    private EditText txtDistractor;
+    private TextView txtResult;
+    private TextView txtStatus;
+    private TextView txtPartialResult;
+    private TextView txtUnsupported;
+    private Button btnStop;
+
+    private ProgressDialog dialog = null;
+
+
+    // 이것의 용도는 대체 무엇일까요...
+    private List<String> finalHyp = new ArrayList<String>();
+
+
+    // 에러 확인 메소드
+    @Override
+    public void rapidSphinxDidStop(String reason, int code) {
+
+
+        if (code == 500) {
+            // 에러
+            System.out.println(reason);
+
+        } else if (code == 522) {
+
+            // 시간 초과 에러
+
+            System.out.println(reason);
+        } else if (code == 200) {
+
+            // 정상 처리 스피치 끝났음
+
+            System.out.println(reason);
+        }
+    }
+
+
+    // 결과 처리
+    @Override
+    public void rapidSphinxFinalResult(String result, List<String> hypArr, List<Double> scores) {
+
+
+        // 소문자 대문자는 영향을 미치지 않게 함.
+        if (result.equalsIgnoreCase(txtWords.getText().toString())) {
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+
+                txtResult.setTextColor(getResources().getColor(android.R.color.holo_green_light, null));
+
+            } else {
+                txtResult.setTextColor(getResources().getColor(android.R.color.holo_green_light));
+
+            }
+
+
+        } else {
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                txtResult.setTextColor(getResources().getColor(android.R.color.holo_red_light, null));
+
+            } else {
+
+                txtResult.setTextColor(getResources().getColor(android.R.color.holo_red_light));
+
+            }
+
+            txtResult.setText(result);
+
+
+
+        }
+
+        if (txtResult.getText().toString() == result) {
```

Comment:
```suggestion
        if (txtResult.getText().toString().equals(result)) {
```
Java는 == 와 equals 의 처리가 다릅니다.
https://coding-factory.tistory.com/536